src/core/Pattern_match.ml

```diff
@@ -78,6 +78,13 @@ type taint_trace_item = {
 type taint_trace = taint_trace_item list [@@deriving show, eq]
 type engine_kind = OSS | Pro [@@deriving show, eq]
 
+type validation_state =
```

Comment:
and also, could we try to find a more descriptive name? "validation state" definitely confused me when i first saw it. would be nice to link it back to its use case

Reply:
I am not going to rename this for now, because it is the same name across semgrep-core, the cli, and the app, and I didn't choose that name, but I did add a comment that should clarify.